odxtools/cli/compare.py

```diff
@@ -27,134 +28,133 @@
+            assert isinstance(service_spec.diag_layer, str)
             rich_print()
             rich_print(
-                f"Changed diagnostic services for diagnostic layer '{service_dict['diag_layer']}' ({service_dict['diag_layer_type']}):"
+                f"Changed diagnostic services for diagnostic layer '{service_spec.diag_layer}' ({service_spec.diag_layer_type}):"
             )
-        if service_dict["new_services"]:
-            assert isinstance(service_dict["new_services"], List)
+        if service_spec.new_services:
+            assert isinstance(service_spec.new_services, List)
             rich_print()
             rich_print(" [blue]New services[/blue]")
-            rich_print(extract_service_tabulation_data(
-                service_dict["new_services"]))  # type: ignore[arg-type]
-        if service_dict["deleted_services"]:
-            assert isinstance(service_dict["deleted_services"], List)
+            rich_print(extract_service_tabulation_data(service_spec.new_services))
+        if service_spec.deleted_services:
+            assert isinstance(service_spec.deleted_services, List)
             rich_print()
             rich_print(" [blue]Deleted services[/blue]")
-            rich_print(extract_service_tabulation_data(
-                service_dict["deleted_services"]))  # type: ignore[arg-type]
-        if service_dict["changed_name_of_service"][0]:
+            rich_print(extract_service_tabulation_data(service_spec.deleted_services))
+        if service_spec.changed_name_of_service[0]:
             rich_print()
             rich_print(" [blue]Renamed services[/blue]")
-            rich_print(extract_service_tabulation_data(
-                service_dict["changed_name_of_service"][0]))  # type: ignore[arg-type]
-        if service_dict["changed_parameters_of_service"][0]:
-            rich_print()
-            rich_print(" [blue]Services with parameter changes[/blue]")
-            # create table with information about services with parameter changes
-            changed_param_column = [
-                str(x) for x in service_dict["changed_parameters_of_service"][
-                    1]  # type: ignore[union-attr]
-            ]
-            table = extract_service_tabulation_data(
-                service_dict["changed_parameters_of_service"][0],  # type: ignore[arg-type]
-                additional_columns=[("Changed Parameters", changed_param_column)])
-            rich_print(table)
-
-            for service_idx, service in enumerate(
-                    service_dict["changed_parameters_of_service"][0]):  # type: ignore[arg-type]
-                assert isinstance(service, DiagService)
+            rich_print(
+                extract_service_tabulation_data([
+                    item for sublist in service_spec.changed_name_of_service for item in sublist
+                    if isinstance(item, DiagService)
+                ]))
```

Comment:
wuff. this is a heavy list comprehension. I would like to see this as "normal" code, i.e. something like
```suggestion
           tmp: List[DiagService] = []
           for sublist in service_spec.changed_name_of_service:
               for item in sublist:
                   if isinstance(item, DiagService):
                       tmp.append(item)
           rich_print(tmp)
```

Reply:
`            tmp: List[DiagService] = []
            for sublist in service_spec.changed_name_of_service:
                for item in sublist:
                    if isinstance(item, DiagService):
                        tmp.append(item)
            rich_print(extract_service_tabulation_data(tmp))`

Made the changes as per the above comment.